src/main/solution_space/path_finder_test_system.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) 2020 Anastasiia Birillo, Elena Lyulina
+
+from __future__ import annotations
+
+import os
+import inspect
+import pkgutil
+import importlib
+from enum import Enum
+from abc import ABCMeta
+from datetime import datetime
+from types import FunctionType
+from typing import Type, TypeVar, List, Dict, Any, Tuple, Optional
+
+from prettytable import PrettyTable, ALL
+
+from src.main.util.consts import EXPERIENCE, LOGGER_NAME
+from src.main.solution_space.serialized_code import Code
+from src.main.solution_space.solution_graph import Vertex
+from src.main.util.file_util import get_class_parent_package
+from src.main.solution_space.consts import TEST_SYSTEM_GRAPH
+from src.main.solution_space.path_finder.path_finder import IPathFinder
+from src.main.solution_space.data_classes import CodeInfo, User, Profile
+from src.main.util.log_util import log_and_raise_error, configure_logger
+from src.main.canonicalization.canonicalization import get_code_from_tree, logging
+from src.main.solution_space.measured_vertex.measured_vertex import IMeasuredVertex
+from src.main.solution_space.solution_space_serializer import SolutionSpaceSerializer
+
+log = logging.getLogger(LOGGER_NAME)
+
+
+class TEST_INPUT(Enum):
+    SOURCE_CODE = 'source'
+    AGE = 'age'
+    EXPERIENCE = 'experience'
+
+
+Class = TypeVar('Class')
+TestInput = Dict[TEST_INPUT, Any]
+# For class(as a key) stores a dict(as a value) with all class methods names(as keys), and class methods(as values)
+MethodsDict = Dict[Type[Class], Dict[str, FunctionType]]
+
+
+# Print results of running all possible PathFinder and MeasuredVertex versions, using https://github.com/kxxoling/PTable
+class TestSystem:
+    _no_method_sign = '---'
+
+    def __init__(self, test_inputs: List[TestInput], serialized_graph_path: str = TEST_SYSTEM_GRAPH):
+        graph = SolutionSpaceSerializer.deserialize(serialized_graph_path)
```

Comment:
Maybe it is better to use **try-catch** block because it can raise an error? Or use it in the SolutionSpaceSerializer.deserialize (sorry, it was my part, but I forgot to change it)